voorhoede/head-start
# feature: basic editable responsive menu

# Changes

- Adds an `AppMenu` to the default layout header. The menu items are editable via the CMS (new models included in migration). The menu has basic responsive behaviour with a full and compact mode. See `AppMenu/README.md` for details.
- Adds icons for the open and close menu buttons.
- Refactors inline logic to small components so they could be reused in multiple places (`AppLogo`, `IconButton`, `SearchLink` and `UnstyledList`).
- Moves components that should only be used in layouts to `src/layouts/`, just like the new `AppMenu`.
- Adds / updates 7 small READMEs.

Only `AppMenu` is really new. All the other changes are assets (icons), refactors and documentation. So the number of files changed may be significant. But the actual changes are not that big.

Note: other types of menu items (like external links) and nested menu items can be added in a separate PR so this one doesn't become any bigger.

# Associated issue

Part of #196.

# How to test

1. Open preview link
2. Verify the header contains the new app menu with items from the CMS
3. Verify the menu items work
4. Verify that when the menu items no longer fit, the menu items are hidden and a menu button appears.
5. Verify the menu button opens the menu in a dialog and you can operate it (focus management, tab and esc work).
6. Change the menu in the CMS
7. Verify the new menu items are rendered (requires local dev server)

# Checklist

- [x] I have made sure that my PR is easy to review (not too big, includes comments) (only `AppMenu` is really new, other files are moved around and many short `README`s are added)
- ~I have added a decision log entry if the change affects the architecture or changes a significant technology~
